apps/server/src/tasks/tasks.service.ts

```diff
@@ -19,6 +20,79 @@ import { Task } from './entities/task.entity';
 
 @Injectable()
 export class TasksService {
+  @Cron('* */1 * * * *')
+  triggerMarkingAsDone() {
+    this.markTasksAsDone();
+  }
+
+  async markTasksAsDone() {
+    const tasks: Task[] = await Task.find({
+      relations: ['chunks', 'chunkInfo'],
+    });
+
+    tasks
+      .filter((task) => task.shouldAutoResolve)
+      .map((task) => {
+        if (task.isFloat) {
+          if (this.isPastDeadline(task)) {
+            // mark task as done (and it's corresponding chunks)
+            Task.update(task.id, {
+              ...task,
+              isDone: true,
+            });
+            task.chunks.map((chunk) => {
+              Chunk.update(chunk.id, {
+                ...chunk,
+                isDone: true,
+              });
```

Comment:
You probably forgot to save an entity there.
But before doing that please consider the amount of tasks we are updating and consider using batching/upsert_all functions.

Reply:
in this case I am using active record update which ensures persistance and additional save is not necessary